scripts/test_workflow_go_pins.py

```
@@ -0,0 +1,112 @@
+            if not asserts:
+                continue
+            checked += 1
+            self.assertTrue(
+                pins, "%s asserts a Go version but pins none exactly" % path.name
+            )
+            self.assertEqual(
+                asserts,
+                pins,
+                "%s pins %s but asserts %s; a half-done bump makes every job "
+                "install one toolchain and then fail its own check"
+                % (path.name, sorted(pins), sorted(asserts)),
```

Ruff (0.16.3)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)